src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -233,4 +237,112 @@ public NeighborNotifyEvent(World world, BlockPos pos, IBlockState state, EnumSet
             return notifiedSides;
         }
     }
+    
+    /**
+     * Fired before a world fires the random tick on the block, registred by the {@link BlockEvent.RandomBlockTickEventRegistry}. 
+     * <br>
+     * This event is called in WorldServer.updateBlocks
+     * <br>
+     * 
+     * If this event is canceled, the block won't do updates, but if you changed the IBlockState variable, the block in the world would be changed to the new one.
+     * <br>
+     * This event is called on the {@link MinecraftForge.#EVENT_BUS}
+     * <br>
+     * 
+     * This event is {@link Cancelable}
+     * 
+     * <br><br>
+     * <strong>!!!WARNING!!! If the block isn't registred by {@link RandomTickEvent.RandomTickEventRegistry}, the event won't be sent!!!</strong>
+     */
+    @Cancelable
+    public static class RandomTickEvent extends BlockEvent
+    {
+    	private IBlockState state;
+    	
+		public RandomTickEvent(World world, BlockPos pos, IBlockState state) {
+			super(world, pos, state);
+			this.state = state;
+		}
+		
+		public void setState(IBlockState state)
+		{
+			this.state = state;
+		}
+		
+		public IBlockState getState()
+		{
+			return state;
+		}
+		
+		/**
+		 * Does the same thing as {@link RandomTickEvent}, but it isn't cancelable and it is called just after the block updated. Sure, if the block has been registred in {@link RandomBlockTickEventRegistry}
+		 * 
+		 *
+		 */
+		public static class Post extends RandomTickEvent
+		{
+			public Post(World world, BlockPos pos, IBlockState state) {
+				super(world, pos, state);
+			}
+		}
+
+		/**
+		 * A class to register {@link BlockEvent.RandomTickEvent} behavior for the block
+		 * 
+		 * <br><br>
+		 * 
+		 * <strong>!!!WARNING!!! Please, don't register too many blocks, because it may reduce Minecraft's performance!</strong><br>
+		 * <strong>Before you register the event for a block, try to look for alternative ways!</strong>
+		 * <br><br>
+		 * 
+		 * <strong> !!!Fires only if the block.getTickRandomly() is true!!!
+		 * 
+		 *
+		 */
+		public static abstract class RandomBlockTickEventRegistry
+		{
+			private RandomBlockTickEventRegistry(){}
+			
+			public static List<Class<? extends Block>> getBlockList() {
+				return blockList;
+			}
+
+			public static void setBlockList(List<Class<? extends Block>> blockList) {
+				RandomBlockTickEventRegistry.blockList = blockList;
+			}
+
+			private static List<Class<? extends Block>> blockList = new ArrayList<Class<? extends Block>>();
+			
+			public static boolean isEventRegistredForBlock(Class<? extends Block> blockClazz)
+			{
+				return blockList.contains(blockClazz);
+			}
+			
+			public static boolean isEventRegistredForBlock(Block block)
+			{
+				return blockList.contains(block.getClass());
+			}
+			
+			public static void registerEventForBlock(Class<? extends Block> blockClazz)
```

Comment:
I would think this name is backwards, as you'd be registering a block for event interaction.